### DISCUSSION TITLE
Is there a reason to why Project64 can't be compiled to run on Linux?

### DISCUSSION THREAD

**Discussion by ghost** [project64/project64#2042] [Support] [answered]
Is it made with like WinForms?

**Comment by project64** [comment, marked as answer]
the core is free of any windows code, i did have it running on android before. but mostly I do not use Linux and no one who has the skills has had the desire to make it work in Linux

**Reply by ghost** [reply]
What has Windows code than?

**Reply by ghost** [reply]
I would like to try porting it to Linux, but it is kinda hard if I don't know where's the Windows code

**Reply by project64** [reply]
the main emulator is Project64 project that is the ui written with window wtl for the UI.  the emulating is in Project64-core this is free of windows code. Plugins like input would have to be designed for the OS. I have plans to redo the UI in ImGui so that it can run on other platforms, but that is way down on my todo list.

**Reply by ghost** [reply]
Where is the UI?
Also good luck with the project, open-source was a good decision

**Reply by ghost** [reply]
Unfortunaly it is harder to read code especially for a big and old project like this, consider porting to linux testing with a VM because there are a lot of people that would like to play N64 on linux, Mupen64 can't play rom hacks because most of them are tested and done on Project64, if you want to play Hack Roms on Linux or Mac there is currently 0 ways to do that. It easer to run a GameCube and Wii game than a n64 game. n64 emulation really is the worst unfortunaly and it would really help to have the linux and mac community to help

**Reply by oddMLan** [reply]
PJ64 also works fine under Wine

**Reply by ghost** [reply]
> PJ64 also works fine under Wine

That should NOT an excuse. People that don't use Windows aren't necessarily tech-savvy

**Reply by ghost** [reply]
should not be* sorry I'm not English

**Reply by bdragoncore** [reply]
WinForms is just something only related to the classic .NET framework.

I agree a linux port would be nice, as I mainly use Linux over Windows these days but it is a process.  You need to decouple the UI from the core, then you need something simple to work first before investing time into writing the entire GUI on top of the emulator.  I am confident I could make a port but certainly not going to happen in a day either, and the main problem is time.
